api/incidents/views.py

```diff
@@ -128,3 +128,28 @@ def retrieve(self, request, pk=None):
         return JsonResponse({"status": 200,
                              "data": serializer.data},
                             status=200)
+
+    def update(self, request, **kwargs):
+        """
+        update:
+        update a redflag.
+        """
+        try:
+            instance = self.get_object()
+
+            if int(instance.createdBy) != request.user.id:
+                return JsonResponse({"status": 403,
+                                     "message": "You can not update a redflag you do not own"},
+                                    status=403)
+
+            request.data['createdBy'] = request.user.id
+            request.data['status'] = 'Draft'
```

Comment:
You can add the logic that checks whether the status has been changed from draft to assert that a user can only update a record if the status is 'draft'.